lib/letsencrypt.js

```diff
@@ -1,215 +1,104 @@
-                                                log.error((new Error('[LEX] SANITY FAIL: no error and yet no certs either')).stack);
-                                            }
-                                            if (certInfo) {
-                                                hostTls = tls.createSecureContext({
-                                                    key:  certInfo.privkey   || certInfo.key,      // privkey.pem
-                                                    cert: certInfo.fullchain || certInfo.cert,     // fullchain.pem (cert.pem + '\n' + chain.pem)
-                                                    ca:   certInfo.ca
-                                                });
-                                                for (var r = 0; r < running.length; r++) {
-                                                    running[r](null, hostTls);
-                                                }
-                                            } else {
-                                                 for (var r = 0; r < running.length; r++) {
-                                                     running[r](null, defaultTls);
-                                                 }
-                                            }
-                                            running = true;
-                                        });
-                                    } else {
-                                        if (running === true) {
-                                            cb(null, defaultTls);
-                                        } else {
-                                            running.push(cb);
-                                        }
-                                    }
-                                    */
-                                }
-                            },
-                            err => {
-                                err && log.error(err);
-                                cb(null, defaultTls);
-                            });
-                        } else {
-                            cb(null, hostTls);
-                        }
+                // prepare domains
+                if (typeof leSettings.domains === 'string') {
+                    leSettings.domains = leSettings.domains.split(',');
+                    for (let d = leSettings.domains.length - 1; d >= 0; d--) {
+                        leSettings.domains[d] = leSettings.domains[d].trim();
+                        // delete empty
+                        !leSettings.domains[d] && leSettings.domains.splice(d, 1);
                     }
-                } else {
-                    cb(null, defaultTls);
                 }
-            };
+                // TODO? Tried adding sites in the gl create but that didn't work so is below with gl.add.
+
+                const pkg = require('../package.json');
+                const gl = require('greenlock').create({
+                    packageRoot: __dirname,
```

Comment:
We should add an option to the leSettings to allow to use `staging` for testing purposes.

Reply:
That's a feature request... wasn't there in the old version ;)

But yes, agreed.

Reply:
BTW, when I say, 'agreed', I mean, I agree it would be useful but do we really need it for an initial release? It would need a setting adding in the admin screen which adds more dependencies.

Reply:
I would make it very simple: add the option in this PR.
Then we can add it in web later (e.g. only as an internal flag without Admin UI).

Reply:
I don't even know how to edit internal flags... must be buried in the doco somewhere ;) If you enlighten me will have a look.

Reply:
Done. Settings are on a per-web adapter basis so needs https://github.com/ioBroker/ioBroker.web/pull/69/commits/d4f041b62ca4a52c44b58f5788467b3cfb197ea2.